manifests/config.pp

```diff
@@ -0,0 +1,94 @@
+# @summary Generates an openssl.conf file using defaults
+#
+# @param ensure
+#   ensure parameter for configfile; defaults to present
+# @param commonname
+#   commonname for config file
+# @param country
+#   value for country
+# @param organization
+#   value for organization
+# @param owner
+#   owner for the configuration file
+# @param group
+#   group for the configuration file
+# @param mode
+#   mode for the configuration file
+# @param state
+#   value for state
+# @param locality
+#   value for locality
+# @param unit
+#   value for unit
+# @param email
+#   value for email
+# @param default_bits
+#   key size to generate
+# @param default_md
+#   message digest to use
+# @param default_keyfile
+#   name for the keyfile
+# @param basicconstraints
+#   version 3 certificate extension basic constraints
+# @param extendedkeyusages
+#   version 3 certificate extension extended key usage
+# @param keyusages
+#   version 3 certificate extension key usage
+# @param subjectaltnames
+#   version 3 certificate extension for alternative names
+#   currently supported are IP (v4) and DNS
+#
+# @example basic usage
+#   openssl::config {'/path/to/openssl.conf':
+#     ensure       => 'present',
+#     commonname   => 'somewhere.org',
+#     country      => 'mycountry',
+#     organization => 'myorg',
+#   }
+#
+define openssl::config (
+  Variant[String, Array[String]] $commonname,
+  String                         $country,
```

Comment:
for all the string parameters, please check if empty strings make sense, otherwise switch to `String[1]`